third_party/amd/lib/TritonAMDGPUTransforms/LowerLoops.cpp

```diff
@@ -127,24 +133,29 @@ ttg::AMDMfmaEncodingAttr getDotEncoding(Value inputValue, unsigned *opIdx,
 // If all the transitive uses of the given value have are used by a convert to
 // the same dot operand encoding, return true and get the shared encoding that
 // needs to be used to be compatible with users' layouts.
-std::optional<ttg::SwizzledSharedEncodingAttr>
-getSharedEncIfAllUsersAreDotEnc(Value loadedValue) {
-  llvm::SmallVector<ttg::SwizzledSharedEncodingAttr> sharedEncs;
+std::optional<ttg::SharedEncodingTrait> getSharedEncIfAllUsersAreDotEnc(
+    Operation *loadOp, tt::ModuleAxisInfoAnalysis &axisInfoAnalysis,
+    const tt::AMD::TargetInfo &targetInfo, bool useAsyncCopy) {
+  assert(loadOp);
+  Value loadedValue = loadOp->getResult(0);
+  llvm::SmallVector<ttg::SharedEncodingTrait> sharedEncs;
   for (Operation *user : loadedValue.getUsers()) {
     LDBG(" getSharedEncIfAllUsersAreDotEnc current user: " << *user);
     if (user->getNumResults() != 1)
       return std::nullopt;
 
-    ttg::SwizzledSharedEncodingAttr tempAttr;
+    ttg::SharedEncodingTrait tempAttr;
     Value userResult = user->getResult(0);
     Type userResType = userResult.getType();
     if (auto memDesc = dyn_cast<ttg::MemDescType>(userResType)) {
       // First time we find a shared encoding in the chain, save it and try to
       // use it if it is compatible with the other users.
-      tempAttr = cast<ttg::SwizzledSharedEncodingAttr>(memDesc.getEncoding());
+      tempAttr = cast<ttg::SharedEncodingTrait>(memDesc.getEncoding());
       // If the immediate user is ttg::LocalAllocOp, likely it's created in
       // TritonAMDGPUOptimizeDotOperands. We should just respect it.
-      if (!getSharedEncIfAllUsersAreDotEnc(userResult).has_value() &&
+      if (!getSharedEncIfAllUsersAreDotEnc(user, axisInfoAnalysis, targetInfo,
+                                           useAsyncCopy)
+               .has_value() &&
```

Comment:
Nit: no need to have `.has_value()` given `std::optional` directly supports `operator bool`. (I'm just a bit annoyed with this extra line for it. ;)